src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -809,6 +1076,1008 @@ static void Test(out System.Collections.Generic.IEnumerable<System.Int32> x)
+
+            compilation.VerifyDiagnostics(
+                // (20,24): error CS8929: Out variable declarations are not allowed within constructor initializers.
+                //         : base(out var x1)
+                Diagnostic(ErrorCode.ERR_OutVarInConstructorInitializer, "x1").WithLocation(20, 24)
+                );
+
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+
+            var x1Decl = tree.GetRoot().DescendantNodes().OfType<ArgumentSyntax>().Where(p => p.Identifier.ValueText == "x1").Single();
+            VerifyModelForOutVar(model, x1Decl);
+        }
+
+        [Fact]
+        public void SimpleVar_14()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        Test2(Test1(out var x1), x1);
+    }
+
+    static object Test1(out dynamic x)
+    {
+        x = 123;
+        return null;
+    }
+
+    static void Test2(object x, object y)
+    {
+        System.Console.WriteLine(y);
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib(text,
+                                                            references: new MetadataReference[] { CSharpRef, SystemCoreRef },
+                                                            options: TestOptions.ReleaseExe,
+                                                            parseOptions: TestOptions.Regular.WithOutVarFeature());
+
+            CompileAndVerify(compilation, expectedOutput: @"123").VerifyDiagnostics();
+
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+
+            var x1Decl = tree.GetRoot().DescendantNodes().OfType<ArgumentSyntax>().Where(p => p.Identifier.ValueText == "x1").Single();
+            var x1Ref = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "x1").Single();
+            VerifyModelForOutVar(model, x1Decl, x1Ref);
+
+            Assert.Null(model.GetAliasInfo(x1Decl.Type));
+            Assert.Equal("dynamic x1", model.GetDeclaredSymbol(x1Decl).ToTestDisplayString());
+        }
+
+        [Fact]
+        public void VarAndBetterness_01()
+        {
+            var text = @"
+public class Cls
+{
+    public static void Main()
+    {
+        Test1(out var x1, null);
+        Test2(out var x2, null);
+    }
+
+    static object Test1(out int x, object y)
+    {
+        x = 123;
+        System.Console.WriteLine(x);
+        return null;
+    }
+
+    static object Test1(out short x, string y)
+    {
+        x = 124;
+        System.Console.WriteLine(x);
+        return null;
+    }
+
+    static object Test2(out int x, string y)
+    {
+        x = 125;
+        System.Console.WriteLine(x);
+        return null;
+    }
+
+    static object Test2(out short x, object y)
+    {
+        x = 126;
+        System.Console.WriteLine(x);
+        return null;
```

Comment:
Is there a test where both overloads have second arg of the same type? (error case)